bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractCaptchaImplV2.java

```diff
@@ -0,0 +1,54 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2023 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import com.adobe.aemds.guide.service.GuideException;
+import com.adobe.cq.forms.core.components.models.form.Captcha;
+import com.adobe.cq.forms.core.components.models.form.FieldType;
+
+/**
+ * Abstract class which can be used as base class for {@link Captcha} implementations.
+ */
+public abstract class AbstractCaptchaImplV2 extends AbstractCaptchaImpl implements Captcha {
```

Comment:
Why is this new AbstractClass implemented ? How is it different from V1 ? If the first one is deprecated, please mention the same in java doc, since this is part of public API surface

Reply:
AbstractCaptchaImplV1 adds the captchaProvider information to the `fd:captcha` custom property by default. Recently, the captchaProvider was made a first-class citizen of the captcha field type JSON. Hence, to keep the JSON clean and avoid copies of the same property, I added a new implementation. 

This is not deprecated as of now, as we need to add recaptcha/hcaptcha v2 version in core components, which will start implementing AbstractCaptchaImplV2 and then we can deprecate this. 

Old Json for captcha, which uses AbstractCaptchaImplV1. (provider is present in "fd:captcha")
```
{
    "id": "hcaptcha-abc574167c",
    "fieldType": "captcha",
    "properties": {
      "fd:dor": {
        "dorExclusion": false
      },
      "fd:path": "/content/hcaptcha",
      "fd:captcha": {
        "provider": "hcaptcha",
        "config": {
          "siteKey": null,
          "uri": null,
          "size": "normal",
          "theme": "light",
          "type": "image"
        }
      },
      "customProp": "customPropValue"
    },
    ":type": "core/fd/components/form/hcaptcha/v1/hcaptcha"
}
```

New Json for captcha type field, which uses AbstractCaptchaImplV2. (captchaProvider is present at top level of json)
```
{
  "id": "turnstile-b4c0808e68",
  "fieldType": "captcha",
  "name": "turnstile1715230058257",
    "fd:captcha": {
      "config": {
        "uri": "https://challenges.cloudflare.com/turnstile/v0/api.js",
        "widgetType": "managed",
        "size": "normal",
        "theme": "light"
      }
    }
  },
  "captchaProvider": "turnstile",
  "captchaDisplayMode": "visible",
  "captchaSiteKey": "siteKey",
  "label": {
    "visible": true,
    "value": "TURNSTILE"
  },
  ":type": "core/fd/components/form/turnstile/v1/turnstile"
}

```

Reply:
Added